### PR TITLE
Added receive mTLS test.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: test/e2e/receive_test.go [hunk 1]
```diff
@@ -715,4 +787,145 @@ func TestReceive(t *testing.T) {
...
+				"prometheus": "prom1",
+				"receive":    "ingestor",
+				"replica":    "0",
+				"tenant_id":  "default-tenant",
+			},
+		})
+	})
+}
+
+func cpyDir(scrDir, dest string) error {
```
Comment (ianbillett) [inline test/e2e/receive_test.go:871]: What does this function do? A recursive copy from `src` to `dest`? Please add a comment

Comment (ianbillett) [reply test/e2e/receive_test.go:871]: Is there not a library function to do this for us?

**Review by ianbillett — APPROVED** [review]
Generally LGTM - nice to see these parts of the code base being exercised :+1: 

My only though would be that we could generate these certificates as part of the e2e setup rather than committing to the repo :man_shrugging: Not a blocker though.

File: test/e2e/certs/create.sh [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  -in e2e_receive_mtls_single_ingestor_client.csr \
+  -CA testca.crt \
+  -CAkey testca.key \
+  -CAcreateserial \
+  -days 99999 \
+  -extfile <(
+    cat <<-EOF
+basicConstraints = CA:FALSE
+nsCertType = client
+nsComment = "OpenSSL Generated Server Certificate"
```
Comment (Namanl2001) [inline test/e2e/certs/create.sh:61]:
```suggestion
nsComment = "OpenSSL Generated Client Certificate"
```

File: test/e2e/certs/create.sh [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  -keyout testca.key \
+  -out testca.crt
+
+openssl genrsa -out testserver.key 2048
+openssl genrsa -out testclient.key 2048
+
+openssl req \
+  -new \
+  -key testserver.key \
+  -subj "/CN=e2e_receive_mtls_single_ingestor-receive-ingestor" \
```
Comment (Namanl2001) [inline test/e2e/certs/create.sh:18]: I believe we can use the same certs for mTLS testing in querier (and maybe in other components too). So why not have some common name for the global cert files 🤔

**Comment by Namanl2001** [comment]
Work in progress

**Pull request closed without merging**